src/components/structures/MatrixChat.js

```diff
@@ -1835,6 +1832,10 @@ export default createReactClass({
             });
         });
 
+        // Create and start the client in the background
+        Lifecycle.setLoggedIn(credentials);
+        await loggedIn;
```

Comment:
The changes are the same, aren't they? Why does it work now but not before?

Reply:
It may not have mattered here, but this version avoid a race where `Lifecycle.setLoggedIn` could have run synchronously and fired `on_logged_in` before we added the listener that waits for it, so we would have gotten. I doubt this happens in real use, but might be possible in tests.